TickSpec/FeatureParser.fs

```diff
@@ -1,92 +1,131 @@
 
-/// Computes combinations of table values
-let internal computeCombinations (tables:Table []) =
-    let values =
-        tables
-        |> Seq.map (fun table ->
-            table.Rows |> Array.map (fun row ->
-                row
-                |> Array.mapi (fun i col ->
-                    table.Header.[i],col
-                )
-            )
-        )
-        |> Seq.toList
-    values |> List.combinations
-
-/// Replace line with specified named values
-let internal replaceLine (xs:seq<string * string>) (scenario,n,tags,line,step) =
-    let replace s =
+/// Parses lines of feature
+let parseFeature (lines:string[]) =
+    let replace combination s =
         let lookup (m:Match) =
             let x = m.Value.TrimStart([|'<'|]).TrimEnd([|'>'|])
-            xs |> Seq.tryFind (fun (k,_) -> k = x)
+            combination |> Seq.tryFind (fun (k,_) -> k = x)
             |> (function Some(_,v) -> v | None -> m.Value)
         let pattern = "<([^<]*)>"
         Regex.Replace(s, pattern, lookup)
-    let step =
-        match step with
-        | GivenStep s -> replace s |> GivenStep
-        | WhenStep s -> replace s |> WhenStep
-        | ThenStep s  -> replace s |> ThenStep
-    let table =
-        line.Table
-        |> Option.map (fun table ->
-            Table(table.Header,
-                table.Rows |> Array.map (fun row ->
-                    row |> Array.map (fun col -> replace col)
-                )
+
+    let computeCombinations examples =
+        let rec combinations source =
+            match source with
+            | [] -> [[]]
+            | (header, rows) :: xs ->
+                [ for row in rows do
+                    for combinedRow in combinations xs ->
+                        (header, row) :: combinedRow ]
+
+        let processRow rowSet =
+            rowSet
+            |> List.fold (fun state (_, (rowMap, rowTags)) ->
+                match state with
+                | None -> None
+                | Some (stateTags, stateMap) ->
+                    let newStateMap =
+                        rowMap
+                        |> Map.fold (fun state key value ->
+                            match state with
+                            | None -> None
+                            | Some map ->
+                                let existingValue = map |> Map.tryFind key
+                                match existingValue with
+                                | None -> Some (map.Add (key, value))
+                                | Some x when x = value -> Some map
+                                | _ -> None) (Some stateMap)
+                    match newStateMap with
+                    | None -> None
+                    | Some s ->
+                        let newStateTags = stateTags @ rowTags
+                        Some (newStateTags, s)
+            ) (Some (List.Empty, Map.empty))
+
+        examples
+        |> Seq.map (fun exampleBlock ->
+            exampleBlock.Table.Header |> List.sort,
```

Comment:
I find the commas on end of lines in the middle of big block somewhat non-intuitive (you have it on more places). https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-tuples requires tuples to be parenthesized. Then to me having contents of the second part huge makes tuple hart to recognize

Reply:
Improved. Do you see any other occurrences you would like to improve?

Reply:
I see that you improved it on other places but I don't see update on this one. Also the guidelines wants the tuples to be parenthesized with only specific exceptions which you didn't change.

Reply:
Resolved, thanks.